pdns/sdig.cc

```diff
@@ -346,7 +349,11 @@ try {
   } else if (tcp) {
     uint16_t counter = 0;
     Socket sock(dest.sin4.sin_family, SOCK_STREAM);
-    sock.connect(dest);
+    sock.setNonBlocking();
+    if (fastopen) {
+      sock.setFastOpenConnect();
+    }
+    sock.connect(dest, INT_MAX);
```

Comment:
That's one huge timeout :) It would be nice to be able to configure it, but for now perhaps we can set it to 10s, as we do for UDP?

Reply:
I did rewrite sdig to use non-blocking mode, so the code is more equal to the rec case. This isn't necessary, but handy for study and diagnosis. But if this code survives a more decent timeout is needed.